authentication.go

```diff
@@ -84,45 +76,40 @@ func authenticate(context context.Context, authservice authClient.AuthServiceCli
 }
 
 func authorize(context context.Context, client *Client, scope string) error {
-	response, err := client.clientsService.IsOwner(context, &clientsClient.IsOwnerRequest{Principal: iam.Credentials.Principal})
-	if err != nil {
-		return fmt.Errorf("%s", fmt.Sprintf("error to verify if user is owner: %s", err.Error()))
-	}
-
-	if response.IsOwner {
-		return nil
-	}
-
-	token, err := generateToken(context, client.authService, scope)
-	if err != nil {
-		return fmt.Errorf("error to generate jwt token: %w", err)
-	}
-
-	authenticated, err := authenticate(context, client.authService, token.Jwt)
-	if err != nil {
-		return fmt.Errorf("error to authenticate: %s", err)
-	}
-	if authenticated.Jwt == "" {
-		return fmt.Errorf("%w", errors.New("you are not allowed to make this request"))
+	resource := strings.Split(scope, ".")[0]
+	// desirableScope := strings.Split(scope, ".")[1]
+	adminScope := fmt.Sprintf("%s.admin", resource)
+	scopes := []string{scope, adminScope, constants.IAMAdmin}
+
+	for _, currentScope := range scopes {
```

Comment:
Pelo o que eu consegui entender, ao invés de verificarmos se quem faz a requisição é owner e, caso não for, ver se ele realmente tem a permissão, vamos lopar esses três(o scopo mesmo, admin de um scopo específico e admin do iam todo) e se algum deles tiver permissão retornar o jwt. 

Porém caso a primeira iteração não tenha permissão, já não daria um erro e quebraria o loop?

Reply:
não pq o que eu verifico é exatamente o oposto, se ele não tem permissão, eu deixo o loop continuar, caso ele saia do loop é pq eu sei que ele não tem nenhuma dessas permissões.